common/meson.build

```
@@ -0,0 +1,89 @@
+install_headers('pkcs11.h', 'pkcs11x.h', subdir: 'p11-kit-1/p11-kit')
+
+libp11_common_sources = [
+  'argv.c', 'argv.h',
+  'attrs.c', 'attrs.h',
+  'array.c', 'array.h',
+  'buffer.c', 'buffer.h',
+  'compat.c', 'compat.h',
+  'constants.c', 'constants.h',
+  'debug.c', 'debug.h',
+  'dict.c', 'dict.h',
+  'hash.c', 'hash.h',
+  'lexer.c', 'lexer.h',
+  'message.c', 'message.h',
+  'path.c', 'path.h',
+  'pkcs11.h', 'pkcs11x.h', 'pkcs11i.h',
+  'runtime.c', 'runtime.h',
+  'url.c', 'url.h',
+  'init.h'
+]
+
+libp11_common = static_library('libp11-common.a', libp11_common_sources,
+                               include_directories: [configinc, commoninc])
+
+libp11_library_sources = [
+  'library.c', 'library.h'
+]
+
+libp11_library = static_library('libp11-library.a', libp11_library_sources,
+                                include_directories: [configinc, commoninc])
+
+libp11_test_sources = [
+  'mock.c', 'mock.h',
+  'test.c', 'test.h'
+]
+
+libp11_test = static_library('libp11-test.a', libp11_test_sources,
+                             include_directories: [configinc, commoninc])
+
+libp11_tool_sources = [
+  'tool.c', 'tool.h'
+]
+
+if host_system != 'windows'
+  libp11_tool_sources += ['unix-peer.c', 'unix-peer.h']
+endif
+
+libp11_tool = static_library('libp11-tool.a', libp11_tool_sources,
+                             include_directories: [configinc, commoninc])
+
+# Tests ----------------------------------------------------------------
+
+common_tests = [
+  'test-tests',
+  'test-compat',
+  'test-hash',
+  'test-dict',
+  'test-array',
+  'test-constants',
+  'test-attrs',
+  'test-buffer',
+  'test-url',
+  'test-path',
+  'test-lexer',
+  'test-message',
+  'test-argv',
+  'test-runtime'
+]
+
+foreach name : common_tests
+  t = executable(name, '@0@.c'.format(name),
+                 c_args: tests_c_args,
+                 include_directories: [configinc, commoninc],
+                 dependencies: libdl_deps,
+                 link_with: [libp11_test, libp11_common])
+  test(name, t)
+endforeach
+
+common_progs = [
+  'frob-getauxval',
+  'frob-getenv'
+]
+
+foreach name : common_progs
+  executable(name, '@0@.c'.format(name),
```

Instead of using string formatting what about using string concatenation here?

```suggestion
  executable(name, name + '.c',
```